src/serac/physics/utilities/functional/domain_integral.hpp

```diff
@@ -227,212 +272,70 @@ void gradient_matrix_kernel(mfem::Vector& K_e, derivatives_type* derivatives_ptr
 {
   using test_element  = finite_element<g, test>;
   using trial_element = finite_element<g, trial>;
-  //  using element_residual_type      = typename trial_element::residual_type;
-  static constexpr int                  test_ndof  = test_element::ndof;
-  static constexpr int                  test_dim   = test_element::components;
-  static constexpr int                  trial_ndof = trial_element::ndof;
-  static constexpr int                  trial_dim  = test_element::components;
-  static constexpr auto                 rule       = GaussQuadratureRule<g, Q>();
-  static constexpr int                  dim        = dimension_of(g);
-  [[maybe_unused]] static constexpr int curl_dim   = dim == 3 ? 3 : 1;
+  static constexpr int   dim        = dimension_of(g);
+  static constexpr int   test_ndof  = test_element::ndof;
+  static constexpr int   test_dim   = test_element::components;
+  static constexpr int   trial_ndof = trial_element::ndof;
+  static constexpr int   trial_dim  = test_element::components;
```

Comment:
I assume this is on purpose that `test_dim` and `trial_dim` are the same?

Reply:
I believe it should be `trial_dim = trial_element::components`.